Sources/Plasma/FeatureLib/pfPython/pyEnum.h

```diff
@@ -46,40 +46,109 @@ You can contact Cyan Worlds, Inc. by email legal@cyan.com
 
 #include <type_traits>
 
+#include "plPythonCallable.h"
 #include "plPythonConvert.h"
 #include "pyObjectRef.h"
 
 // Helper functions used by the enum glue
 class pyEnum
 {
 public:
-    // FIXME This should probably be integrated into plPythonConvert, but I can't get it to work right now...
-    template<typename T, typename R>
-    static R GetUnderlyingEnumValue(T value)
+    class EnumValue
     {
-        return value;
+        pyObjectRef fName;
+        pyObjectRef fValue;
+
+    public:
+        EnumValue() = delete;
+        EnumValue(const EnumValue&) = delete;
+
+        template <size_t _Sz, typename _ValueT>
+        EnumValue(const char (&name)[_Sz], _ValueT&& value)
+            : fName(PyUnicode_FromStringAndSize(name, _Sz - 1)),
+              fValue(plPython::ConvertFrom(static_cast<std::underlying_type_t<_ValueT>>(value)))
+        {
+            static_assert(std::is_enum_v<_ValueT>, "EnumValue can only be used with enumerations");
+            static_assert(std::is_integral_v<std::underlying_type_t<_ValueT>>, "EnumValue can only be used with enumerations with integral underlying types");
+        }
+
+        EnumValue(EnumValue&& move) noexcept
+            : fName(std::move(move.fName)),
+              fValue(std::move(move.fValue))
+        {
+        }
+
+        PyObject* ReleaseName() { return fName.Release(); }
+        PyObject* ReleaseValue() { return fValue.Release(); }
+    };
+
+private:
+    template <typename Arg>
+    static inline void InsertEnumValue(PyObject* list, Py_ssize_t& i, Arg& arg)
```

Comment:
I un-templated this helper method and declared `arg` directly as `EnumValue&`. As far as I can tell, this doesn't need to be a template, because it's not variadic and `Arg` can only ever be `EnumValue`. Let me know if there was another reason for this being a template that I missed.

Reply:
Untemplating this method is fine, I think. I copied and pasted a lot of this from the code I just wrote for `KwArg` where having a templated method was useful for emitting good error messages when a keyword argument appeared unexpectedly. Here, we're relying on macros pretty heavily, so having human-readable errors isn't as important.